lib/DAPIClient.js

```diff
@@ -0,0 +1,57 @@
+const GrpcTransport = require('./transport/GrpcTransport');
+const JsonRpcTransport = require('./transport/JsonRpcTransport/JsonRpcTransport');
+
+const CoreMethodsFacade = require('./methods/core/CoreMethodsFacade');
+const PlatformMethodsFacade = require('./methods/platform/PlatformMethodsFacade');
+
+const createAddressProviderFromOptions = require('./addressProvider/createAddressProviderFromOptions');
+const requestJsonRpc = require('./transport/JsonRpcTransport/requestJsonRpc');
+
+class DAPIClient {
+  /**
+   * @param {DAPIClientOptions} [options]
+   */
+  constructor(options = {}) {
+    this.options = {
+      timeout: 2000,
+      retries: 3,
+      ...options,
+    };
+
+    this.addressProvider = createAddressProviderFromOptions(this.options);
```

Comment:
Since we have addresses in terms of crypto addresses, maybe it's better to rename it to something like `mnHostProvider`?

Reply:
We could rename it dapiAddressProvider